### PR TITLE
Mailbox 269 implement Attachment capability

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: server/container/guice/cassandra-guice/src/test/java/org/apache/james/JamesCapabilitiesServerTest.java [hunk 1]
```diff
@@ -70,21 +70,26 @@ public void startShouldSucceedWhenMoveCapability() throws Exception {
         server = createCassandraJamesServer(mailboxManager);
 
         server.start();
+    }
 
+    @After
+    public void teardown() {
         server.stop();
```
Comment (aduprat) [inline server/container/guice/cassandra-guice/src/test/java/org/apache/james/JamesCapabilitiesServerTest.java]: What appens if server is `null`?

Comment (mbaechler) [inline server/container/guice/cassandra-guice/src/test/java/org/apache/james/JamesCapabilitiesServerTest.java]: it should not happen, so the test fails, but anyway, there's nothing to clean up, really

File: mailbox/api/src/main/java/org/apache/james/mailbox/MessageManager.java [hunk 2]
```diff
@@ -41,8 +41,12 @@
  * 
  * A {@link MessageManager} should be valid for the whole {@link MailboxSession}
  */
-public interface MessageManager {
+public interface MessageManager extends CapabilityAware<MessageManager.Capabilities> {
 
+    enum Capabilities {
+        Attachment
```
Comment (rouazana) [inline mailbox/api/src/main/java/org/apache/james/mailbox/MessageManager.java]: Attachments?

File: mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java [hunk 3]
```diff
@@ -67,16 +67,14 @@
  * </p>
  */
 
-public interface MailboxManager extends RequestAware, MailboxListenerSupport {
+public interface MailboxManager extends RequestAware, MailboxListenerSupport, CapabilityAware<MailboxManager.Capabilities> {
 
     enum Capabilities {
         Basic,
```
Comment (rouazana) [inline mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java]: Has Basic still an interest if you can return EnumSet.noneOf...?

Comment (mbaechler) [inline mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java]: probably not

File: mailbox/api/src/main/java/org/apache/james/mailbox/MessageManager.java (hunk 2 above)
Comment (mbaechler) [inline mailbox/api/src/main/java/org/apache/james/mailbox/MessageManager.java]: Move is not a plural form, it's consistent, don't you think ?

Comment (rouazana) [inline mailbox/api/src/main/java/org/apache/james/mailbox/MessageManager.java]: Move is an action. Userflags is plurial.

Comment (mbaechler) [inline mailbox/api/src/main/java/org/apache/james/mailbox/MessageManager.java]: ok, userflags is a plural ...

File: server/container/guice/guice-common/src/main/java/org/apache/james/jmap/JMAPModule.java [hunk 4]
```diff
@@ -85,19 +86,23 @@ private PropertiesConfiguration getConfiguration(FileSystem fileSystem) throws F
...
     @Singleton
-    public static class MoveCapabilityPrecondition implements ConfigurationPerformer {
+    public static class RequiredCapabilitiesPrecondition implements ConfigurationPerformer {
 
         private final MailboxManager mailboxManager;
+        private final MessageManager messageManager;
 
         @Inject
-        public MoveCapabilityPrecondition(MailboxManager mailboxManager) {
+        public RequiredCapabilitiesPrecondition(MailboxManager mailboxManager, MessageManager messageManager) {
```
Comment (rouazana) [inline server/container/guice/guice-common/src/main/java/org/apache/james/jmap/JMAPModule.java]: Are you sure MessageManager can be injected? That's not something you get from a MailboxManager instance?

Comment (mbaechler) [inline server/container/guice/guice-common/src/main/java/org/apache/james/jmap/JMAPModule.java]: no, Manager are top level services I think

Comment (rouazana) [inline server/container/guice/guice-common/src/main/java/org/apache/james/jmap/JMAPModule.java]: `    @Override
    protected StoreMessageManager createMessageManager(Mailbox mailboxRow, MailboxSession session) throws MailboxException {
        return new CassandraMessageManager(getMapperFactory(),
            getMessageSearchIndex(),`

Comment (rouazana) [inline server/container/guice/guice-common/src/main/java/org/apache/james/jmap/JMAPModule.java]: (in CassandraMailboxManager)

Comment (mbaechler) [inline server/container/guice/guice-common/src/main/java/org/apache/james/jmap/JMAPModule.java]: well ... we can't use MessageManager for that then as it's bound to a session.
I'll try to put it back into MailboxManager then. WDYT ?

Comment (rouazana) [inline server/container/guice/guice-common/src/main/java/org/apache/james/jmap/JMAPModule.java]: Yes that was my point, I think we should keep capabilities in MailboxManager for now.

**Comment by chibenwa** [comment]
:+1:

**Comment by aduprat** [comment]
:+1:

**Comment by rouazana** [comment]
:+1:

**Comment by mbaechler** [comment]
test this please

**Comment by mbaechler** [comment]
merged

**Pull request closed without merging**